### DISCUSSION TITLE
Showing legend

### DISCUSSION THREAD

**Discussion by Ahmad-Abdou** [gosling-lang/gosling.js#1085] [Q&A] [answered]
Is there a way to implement legend in this library, will this functionality be available in the future?

**Comment by sehilyi** [comment]
There are several options for legend. You can search for a keyword "legend:" http://gosling-lang.org/docs/reference

For example, to show the legend for the color, you can use `color: { ..., legend: true }`

**Reply by Ahmad-Abdou** [reply]
The thing is that I wanted to show axes labels
Currently, this is how I am implementing it
// Track specification used in GUI
const y_col_preset = ""
const x_col_preset = ""

const trackTemplate = {
    "data": {
        "url": "",
        "type": "",
        "separator": "",
        "column": x_col_preset,
        "value": y_col_preset,
        "binSize": 10,
        "sampleLength": 1000,
    },
    "metadata": {"categories": ["Track"]},
    "mark": "point",
    "x": { "field": x_col_preset, "type": "genomic", "axis": "bottom", "legend": true, "linkingId": "detail",},
    "y": { "field": y_col_preset, "type": "quantitative", "axis": "left", "domain": [0, 1], "baseline": "2", "legend": true},
    "color": { "value": "orange", "title": "Trackss", "legend": true},
    "opacity": { "value": 0.8 },
    "size": { "value": 3 },
    "tooltip": [
        { "field": y_col_preset, "type": "quantitative", "format": "0.2f", "alt": y_col_preset },
        { "field": x_col_preset, "type": "genomic", "format": "0.2f", "alt": x_col_preset }
    ],
};

export { trackTemplate };


and this is how it is displayed
![image](https://github.com/user-attachments/assets/a745f0ca-70cb-4162-8804-9a7df4262ab9)

**Reply by sehilyi** [reply, marked as answer]
Hi, do you mean you want to show the field name in the x and y axes, e.g., "y_col_preset"? If so, it is not supported directly, but you could use the 'title' property of the track to show such information:

![Screenshot 2024-07-25 at 9 40 57 AM](https://github.com/user-attachments/assets/6554ff90-fc92-42aa-8669-b6c9645525ab)

```
{
          "title": "[Y-AXIS-NAME AND/OR X-AXIS_NAME]",
          "data": {
                    ...
```

**Reply by Ahmad-Abdou** [reply]
Thanks for helping out
